builtin/logical/database/path_roles.go

```diff
@@ -633,8 +638,41 @@ func (b *databaseBackend) pathStaticRoleCreateUpdate(ctx context.Context, req *l
 		}
 	}
 
+	dbConfig, err := b.DatabaseConfig(ctx, req.Storage, role.DBName)
+	if err != nil {
+		return nil, err
+	}
+
+	lastVaultRotation := role.StaticAccount.LastVaultRotation
+	if passwordRaw, ok := data.GetOk("password"); ok {
+		// We will allow users to update the password until the point where
+		// Vault assumes management of the account so that we don't break the
+		// promise of Vault being the source of truth.
+		updateAllowed := lastVaultRotation.IsZero() || lastVaultRotation.After(time.Now())
+		if updateAllowed {
+			role.StaticAccount.Password = passwordRaw.(string)
+			if dbConfig.ConnectionDetails["self_managed"].(bool) {
+				// continue to support the deprecated self_managed_password field
+				role.StaticAccount.SelfManagedPassword = passwordRaw.(string)
+			}
+		} else {
+			return logical.ErrorResponse("update not allowed after rotation", "role", name, "lastVaultRotation", lastVaultRotation), nil
```

Comment:
nit: should we make the error message clearer to state that the `password` can't be updated? Since the static role itself is technically still updateable
```suggestion
			return logical.ErrorResponse("updating password not allowed after rotation", "role", name, "lastVaultRotation", lastVaultRotation), nil
```